dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemoryStore.cs

```diff
@@ -0,0 +1,505 @@
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore"/> backed by a Azure CosmosDB Mongo vCore database.
+/// Get more details about Azure Cosmos Mongo vCore vector search  https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search
+/// </summary>
+public class AzureCosmosDBMongoDBMemoryStore : IMemoryStore, IDisposable
+{
+    private readonly MongoClient _mongoClient;
+    private readonly IMongoDatabase _mongoDatabase;
+    private readonly AzureCosmosDBMongoDBConfig _config;
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBMongoDBMemoryStore instance using a Azure CosmosDB Mongo vCore connection string
+    /// and other properties required for vector search.
+    /// </summary>
+    /// <param name="connectionString">Connection string required to connect to Azure Cosmos Mongo vCore.</param>
+    /// <param name="databaseName">Database name for Mongo vCore DB</param>
+    /// <param name="config">Azure CosmosDB MongoDB Config containing specific parameters for vector search.</param>
+    public AzureCosmosDBMongoDBMemoryStore(
+        string connectionString,
+        string databaseName,
+        AzureCosmosDBMongoDBConfig config
+    )
+    {
+        MongoClientSettings settings = MongoClientSettings.FromConnectionString(connectionString);
+        this._config = config;
+        settings.ApplicationName = this._config.ApplicationName;
+        this._mongoClient = new MongoClient(settings);
+        this._mongoDatabase = this._mongoClient.GetDatabase(databaseName);
+    }
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBMongoDBMemoryStore instance using a Azure CosmosDB MongoDB client
+    /// and other properties required for vector search.
+    /// </summary>
+    public AzureCosmosDBMongoDBMemoryStore(
+        IMongoClient mongoClient,
+        string databaseName,
+        AzureCosmosDBMongoDBConfig config
+    )
+    {
+        MongoClientSettings settings = mongoClient.Settings;
+        this._config = config;
+        settings.ApplicationName = this._config.ApplicationName;
+        this._mongoClient = new MongoClient(settings);
+        this._mongoDatabase = this._mongoClient.GetDatabase(databaseName);
+    }
+
+    /// <inheritdoc/>
+    public async Task CreateCollectionAsync(
+        string collectionName,
+        CancellationToken cancellationToken = default
+    )
+    {
+        await this
+            ._mongoDatabase.CreateCollectionAsync(
+                collectionName,
+                cancellationToken: cancellationToken
+            )
+            .ConfigureAwait(false);
+        var indexes = await this.GetCollection(collectionName)
+            .Indexes.ListAsync(cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+
+        if (
+            !indexes
+                .ToList(cancellationToken: cancellationToken)
+                .Any(index => index["name"] == this._config.IndexName)
+        )
```

Comment:
The code formatting is not aligned with code formatting in repo. We don't keep parentheses in separate line. Also, if condition is complex, it's better to define separate variable for it and then re-use it in `if` statement.